R/loadFromMothur.R

```diff
@@ -0,0 +1,206 @@
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{makeTreeSummarizedExperimentFromDADA2}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{loadFromQIIME2}}
+#'
+#' @export
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @references
+#' \url{https://mothur.org/}
+#' 
+#' @examples
+#' # Abundance table
+#' counts <- "~/Desktop/Baxter_FITs_Microbiome_2016_fit.final.tx.1.subsample.shared"
+#' # Taxa table
+#' taxa <- "~/Desktop/Baxter_FITs_Microbiome_2016_fit.final.tx.1.cons.taxonomy"
+#' # Sample meta data
+#' meta <- "~/Desktop/Baxter_FITs_Microbiome_2016_mapping.csv"
+#' 
+#' # Creates tse object from files
+#' \dontrun{tse <- loadFromMothur(counts, taxa, meta)}
+#' 
+
+loadFromMothur <- function(featureTableFile,
+                           taxonomyTableFile = NULL,
+                           sampleMetaFile = NULL,
+                           featureNamesAsRefSeq = TRUE,
+                           refSeqFile = NULL,
+                           phyTreeFile = NULL,
+                           ...) {
+  
+    # input check
+    if(!.is_non_empty_string(featureTableFile)){
+      stop("'featureTableFile' must be a single character value.",
+           call. = FALSE)
+    }
+    if(!is.null(taxonomyTableFile) && !.is_non_empty_string(taxonomyTableFile)){
+      stop("'taxonomyTableFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    if(!is.null(sampleMetaFile) && !.is_non_empty_string(sampleMetaFile)){
+      stop("'sampleMetaFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    if(!.is_a_bool(featureNamesAsRefSeq)){
+      stop("'featureNamesAsRefSeq' must be TRUE or FALSE.", call. = FALSE)
+    }
+    if(!is.null(refSeqFile) && !.is_non_empty_string(refSeqFile)){
+      stop("'refSeqFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    if(!is.null(phyTreeFile) && !.is_non_empty_string(phyTreeFile)){
+      stop("'phyTreeFile' must be a single character value or NULL.",
+           call. = FALSE)
+    }
+    #
+    
+    # Reads the featureTablefile 
+    feature_tab <- .read_mothur_feature(featureTableFile, ...)
+    
+    # If rowData information exists, gets that. Otherwise, tax_tab is just data frame without information
+    if (!is.null(taxonomyTableFile)) {
+      taxa_tab <- .read_mothur_taxonomy(taxonomyTableFile, feature_tab, ...)
+    } else {
+      taxa_tab <- S4Vectors:::make_zero_col_DataFrame(nrow(feature_tab))
+    }
+    
+    # If colData information exists, gets that. Otherwise, sample_tab is just data frame without information
+    if (!is.null(sampleMetaFile)) {
+      sample_meta <- .read_mothur_sample_meta(sampleMetaFile, feature_tab)
+    } else {
+      sample_meta <- S4Vectors:::make_zero_col_DataFrame(ncol(feature_tab))
+    }
+    
+    if (!is.null(phyTreeFile)) {
+      tree <- .read_qza(phyTreeFile, ...)
```

Comment:
This is not applicable here, since the `.read_qza` function is `QIIME2` output specific. You probably just need to add another function as warpper for `ape::read.tree`